homeassistant/components/hikvision/camera.py

```diff
@@ -0,0 +1,183 @@
+"""Support for Hikvision camera streams."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import httpx
+
+from homeassistant.components.camera import Camera, CameraEntityFeature
+from homeassistant.components.stream import CONF_RTSP_TRANSPORT, RTSP_TRANSPORTS
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.httpx_client import get_async_client
+
+from . import HikvisionConfigEntry
+from .const import CONF_RTSP_PORT, DEFAULT_RTSP_PORT, DOMAIN
+from .helpers import HikvisionChannel, build_rtsp_url, build_snapshot_url
+
+_LOGGER = logging.getLogger(__name__)
+
+# Timeout for fetching snapshot images
+GET_IMAGE_TIMEOUT = 10
+
+# Use sub-stream (lower resolution) for snapshots to reduce load
+SNAPSHOT_STREAM_TYPE = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: HikvisionConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up Hikvision cameras from a config entry."""
+    data = entry.runtime_data
+    channels = data.channels
+
+    if not channels:
+        _LOGGER.debug("No video channels found on device")
+        return
+
+    async_add_entities(
+        HikvisionCamera(entry=entry, channel=channel)
+        for channel in channels
+        if channel.enabled
+    )
+
+
+class HikvisionCamera(Camera):
+    """Representation of a Hikvision camera stream."""
+
+    _attr_has_entity_name = True
+    _attr_supported_features = CameraEntityFeature.STREAM
+
+    def __init__(
+        self,
+        entry: HikvisionConfigEntry,
+        channel: HikvisionChannel,
+    ) -> None:
+        """Initialize the camera."""
+        super().__init__()
+        self._entry = entry
+        self._data = entry.runtime_data
+        self._channel = channel
+
+        # Build unique ID
+        self._attr_unique_id = f"{self._data.device_id}_camera_{channel.id}"
+
+        # Build entity name based on device type and channel count
+        if self._data.device_type == "NVR" or len(self._data.channels) > 1:
+            self._attr_name = channel.name
+        else:
+            # Single camera - use None to just use device name
+            self._attr_name = None
```

Comment:
What is a channel exactly? Is a channel a separate camera in a separate room? Should they be their own device?